core/domain/state_domain.py

```diff
@@ -1599,13 +1599,12 @@ def update_interaction_default_outcome(self, default_outcome_dict):
             old_content_id_list.append(
                 self.interaction.default_outcome.feedback.content_id)
 
-        if default_outcome_dict:
```

Comment:
@puneeth2001, should we handle this before creating the domain object out of the dict? 

@ankita240796, Thanks for checking this, I think earlier we were validating this before calling the from_dict method and we should do the same. (In general, from_dict should always accept a dict and should only return the expected class object.)